### ISSUE TITLE
[Enhancement] Specify Field(s) To Be Imported

### ISSUE THREAD

**Issue by edolyne** [EllisLab/PlayaMatrixImporter#4]
We manage a large EE install where importing all of our Matrix and Playa fields at one time locks up our database.  It would be great to have an option on the import screen to be able to select all, or specific fields to import at a given time.

**Comment by kevincupp** [comment]
Thanks for the suggestion! Are you saying it locks up the tables preventing site visitors from accessing he site? Are you able to do the import on another environment, and then re-import the database from that environment?

**Comment by roberthallatt** [comment]
I just gave this a shot and I received the following MySQL error:

A Database Error Occurred
Error Number: 1117
Too many columns
ALTER TABLE `exp_channel_data` ADD `field_id_1745` VARCHAR(8)
Filename: libraries/api/Api_channel_fields.php
Line Number: 675

Our DB is pretty big. Any thoughts about how I might overcome this?

**Comment by kevincupp** [comment]
😬 Sorry to hear, Robert! Unfortunately MySQL has a pretty hard limit on MySQL row sizes. We're aware of this and will eventually move away from this kind of limiting schema with ExpressionEngine, but I don't think there's currently much you can do but try to get rid of some fields, either by just doing without them or finding a way to reuse fields across channels.

**Comment by roberthallatt** [comment]
Yeah, I thought that would be the answer. I will be recoding our site using Stash so I think I can dramatically cut down on the number of playa fields that are used in every single channel. I’m just not looking forward to having to reset the content in these fields for 1500 web pages.

**Comment by goodtimeclub** [comment]
For anyone else looking to get around this limitation, it can be done by modifying the get_{fieldtype}_fields() function in **libraries/playa_importer.php** and **libraries/matrix_importer.php**. _(See example below)_
Simply specify which fields you'd like to migrate, run the importer, then delete the original fields from the system. It means stepping lightly and slowly but _will_ get you to the finish line 😉

Line 113 of **libraries/playa_importer.php**:
`public function get_playa_fields()
{
	return ee()->db->where('field_type','playa')
		->where_in('field_id',array(1134,1135))
		->get('channel_fields')
		->result_array();
}`
